app/src/main/java/com/wafflestudio/siksha2/ui/main/community/PostCreateViewModel.kt

```diff
@@ -0,0 +1,211 @@
+package com.wafflestudio.siksha2.ui.main.community
+
+import android.content.Context
+import android.net.Uri
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.wafflestudio.siksha2.models.Board
+import com.wafflestudio.siksha2.models.Post
+import com.wafflestudio.siksha2.repositories.CommunityRepository
+import com.wafflestudio.siksha2.utils.ImageUtil
+import com.wafflestudio.siksha2.utils.showToast
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import okhttp3.MediaType.Companion.toMediaTypeOrNull
+import okhttp3.MultipartBody
+import okhttp3.RequestBody.Companion.asRequestBody
+import okhttp3.RequestBody.Companion.toRequestBody
+import java.net.URL
+import javax.inject.Inject
+
+@HiltViewModel
+class PostCreateViewModel @Inject constructor(
+    savedStateHandle: SavedStateHandle,
+    private val communityRepository: CommunityRepository
+) : ViewModel() {
+    private val _boards = MutableStateFlow<List<Board>>(listOf())
+    val boards: StateFlow<List<Board>> = _boards
+
+    private val _board = MutableStateFlow<Board>(Board())
+    val board: StateFlow<Board> = _board
+
+    private val _post = MutableStateFlow<Post>(Post())
+    val post: StateFlow<Post> = _post
+
+    private val _title = MutableStateFlow<String>("")
+    val title: StateFlow<String> = _title
+
+    private val _content = MutableStateFlow<String>("")
+    val content: StateFlow<String> = _content
+
+    private val _imageUriList = MutableStateFlow<List<Uri>>(emptyList())
+    val imageUriList: StateFlow<List<Uri>> = _imageUriList
+
+    private val _imageFileList = MutableStateFlow<Map<String, ByteArray>>(emptyMap())
+    val imageFileList: StateFlow<Map<String, ByteArray>> = _imageFileList
+
+    private val _createPostState = MutableStateFlow(CreatePostState.USER_INPUT)
+    val createPostState: StateFlow<CreatePostState> = _createPostState
+
+    private val _postId = MutableStateFlow<Long>(-1)
+    val postId: StateFlow<Long> = _postId
```

Comment:
~~postId랑 boardId SavedStateHandle.getStateFlow로 하는거어때? 잘 동작하는지는 확인안해봤지만 코드이뻐질듯~~

```suggestion
    private val _postId = savedStateHandle.getStateFlow("post_id", -1)
```

https://developer.android.com/topic/libraries/architecture/viewmodel/viewmodel-savedstate#savedstate-stateflow

Reply:
아 postId는 게시글 생성하는 경우에 서버 응답으로 얻은 id구나 생성하기전엔 id가 없으니까

이름 수정하는거어때? createdPostId 이런식으로

Reply:
뷰모델 하나가 생성도 수정도 하려니 빡세네 그냥 분리할걸그랬나....모르겠다